crates/core/src/params.rs

```diff
@@ -84,3 +84,76 @@ impl From<libsql_sys::Value> for Value {
         }
     }
 }
+
+// Heavily inspired by rusqlite's ValueRef
+pub enum ValueRef<'a> {
+    Null,
+    Integer(i64),
+    Real(f64),
+    Text(&'a [u8]),
```

Comment:
Minor but we could leave this as a `str` to keep the enforced utf8 since its trivial to convert back into a u8.

Reply:
fair enough, rusqlite was like that probably mostly to satisfy weird windows encodings, but we're more opinionated on that front